### ISSUE TITLE
local development: constant network error / reconnecting

### ISSUE THREAD

**Issue by lw1990** [RoepStoep/lidraughts#5]
I followed the windows onboarding instructions for lichess found at https://github.com/ornicar/lila/wiki/Lichess-Development-Onboarding-(Windows), with the following modifications for lidraughts to work on my local machine. The problem is, even though I can create matches and everything compiles/runs fine, I get constant network error / reconnecting status in the browser, resulting in not being able to make any moves or sign in or register. I think it's because I missed some location where lidraughts is still trying to use a web host rather than a localhost. 

Is it even easy to set this project up for development with a local host or should it be set up with a web host? If it's something easy I missed I would really appreciate any help.

copied `bin\gen\geoip.ps1` from `lila` to `lidraughts` directory before running `powershell -executionpolicy bypass -File bin\gen\geoip.ps1` because the lidraughts repo did not have the file included

modified `nginx.conf` block to be the following:
`	server {
	  server_name lidraughts-assets.local;
	  listen 80;
	  charset utf-8;
	  location /assets {
		add_header "Access-Control-Allow-Origin" "*";
		alias "C:\Users\Luke\Documents\git\lidraughts2\lidraughts\public";
	  }
    }`

modified `base.conf` `net` block to be the following:
`net {
  domain = "localhost:9663"
  protocol = "http://"
  base_url = ${net.domain}
  ip = "94.214.154.195"
  asset {
    domain = "lidraughts-assets.local"
    version = 10
  }
  email = "contact@lidraughts.org"
  crawlable = false
}`

modified `hosts` (c:\Windows\System32\Drivers\etc\hosts) to include `127.0.0.1 lidraughts-assets.local`

**Comment by lw1990** [comment]
I modified `base.conf` further to match the one from `lila` more closely, but still no luck, still getting that network disconnected / reconnecting message at the top right in the browser (image below)

The `base.conf` `net` block has the lidraughts default in this repo shown with `#commented` lines for reference to what was modified

> net {
#domain = "lidraughts.org"
  domain = "localhost:9663"
#socket.domain line is not in the lidraughts base.conf
  socket.domain = ${net.domain}
  protocol = "http://"
  base_url = ${net.protocol}${net.domain}
#ip = "94.214.154.195" (not sure if changing to the lila ip below helped or hurt?)
  ip = "5.196.91.160"
  asset {
    #domain = ${net.domain}
    domain = "lidraughts-assets.local"
    version = 10
  }
  email = "contact@lidraughts.org"
  crawlable = false
}

**Comment by RoepStoep** [comment]
If it keeps reconnecting there's a problem with the socket. If you look at the developer tools in your browser you must be able to spot the failing connection. You will probably see that it tries to connect to socket.lidraughts.org or the equivalent url of your setup. Check if that is routed correctly in your config to localhost::9663

I see the lichess onboarding instructions have changed since the ones I used though. If you can't get it to work that way you might want to try the previous version at https://github.com/ornicar/lila/wiki/Lichess-Development-Onboarding-(Windows)/3ffa2a6a9eed421c73f32e61eec0175d870b5d17

It uses a different way to set things up, which is what I still use for development, so that should work for sure. It looks like you'd only have to change your nginx.conf and hosts file, and update net.domain in your base.conf or application.conf

**Comment by lw1990** [comment]
Thanks, there is indeed a bunch of failing network 404's in the network tab of the developer tools as it tries to connect even after I made those latest changes to `nginx.conf`, `hosts`, `application.conf` and `base.conf` per the previous version of https://github.com/ornicar/lila/wiki/Lichess-Development-Onboarding-(Windows)/3ffa2a6a9eed421c73f32e61eec0175d870b5d17

https://i.imgur.com/5HHYD3e.png

could you upload your local nginx.conf, application.conf, and base.conf if it's not too much trouble so I can compare with mine? Once I figure this out the first time I'm going to write a guide for the wiki here to remind myself and others how to get it to work.

[confs.zip](https://github.com/RoepStoep/lidraughts/files/2298678/confs.zip)

**Comment by RoepStoep** [comment]
I uploaded the nginx.conf and application.conf I use in development, just replace yourdomain.org and path_to_lidraughts_root with your values. I use the exact base.conf as it is in the repo.

But your confs look fine, did you add both l.org and socket.l.org to your hosts file like this? 
127.0.0.1 l.org socket.l.org

From your screenshot it looks like it is trying to connect to socket.localhost:9663 etc, did you maybe open localhost:9663 in your browser? Then you should navigate to l.org, which should make the socket connect to socket.l.org, which is routed to the server at localhost:9663 in the nginx conf

**Comment by lw1990** [comment]
I did add that to my hosts file previously, and yes I opened localhost:9663 in the browser, opening l.org:9663 in or similar is the only way I can get it to show the lidraughts dashboard (but it does have network errors)
If I just open l.org or yourdomain.org or similar it just shows me the nginx further configuration required page https://i.imgur.com/kxsD3HL.png

I don't know why it would do this since I copied/pasted your nginx.conf file though

**Comment by lw1990** [comment]
Here are the complete steps I'm taking (after installing all prerequisites from https://github.com/ornicar/lila/wiki/Lichess-Development-Onboarding-(Windows)/3ffa2a6a9eed421c73f32e61eec0175d870b5d17)

0. restart PC and delete any existing lidraughts files to remove as many variables as possible
1. open cmd.exe
2. `cd C:\`
3. `git config --global core.autocrlf false`
4. `git clone --recursive https://github.com/RoepStoep/lidraughts.git`
5. `git config --global core.autocrlf true`
6. `cd lidraughts`
7. create `conf/application.conf` with the following content https://github.com/RoepStoep/lidraughts/blob/master/doc/example_application.conf
8. back in cmd.exe in the lidraughts dir, `yarn install`
9. `.\ui\build.bat`
10. `.\bin\dev.bat compile`
11. edit hosts file to include `127.0.0.1 lidraughts-assets.local`
`127.0.0.1 l.org socket.l.org en.l.org`
`127.0.0.1 yourdomain.org socket.yourdomain.org`
12. edit my `nginx.conf` to be the following: https://github.com/RoepStoep/lidraughts/blob/master/doc/example_nginx.conf
13. replace `path_to_lidraughts_root/public` in `nginx.conf` with my local path to `lidraughts/public` which is `alias "C:\lidraughts\public";` and `root"C:\lidraughts\public";` (does double quoted backslash paths work here? that is the most likely culprit I can think of)
14. restart (or start) nginx.exe (`cd C:\nginx-1.15.2` -> `start nginx`)
15. back in cmd.exe, start MongoDB with `mongod`
16. `cd C:\lidraughts`
17. `.\bin\dev.bat`
18. `run 9663`
19. navigate to `l.org` in browser -> navigate to `yourdomain.org` in separate browser tab

**Comment by lw1990** [comment]
Small update:
I finally got l.org in a browser to show the lidraughts page instead of the nginx welcome page. The problem was my hosts file had too much in it - either the `127.0.0.1 yourdomain.org socket.yourdomain.org` or `127.0.0.1 lidraughts-assets.local` lines were causing problems. Once I removed those I could load up l.org and see the dashboard, but I'm still getting the network disconnection problem. Now the error codes are 403 forbidden. Sigh.
